File: java/src/main/java/fr/sdecout/handson/rest/author/AuthorResponse.java
package fr.sdecout.handson.rest.author;

import fr.sdecout.handson.rest.shared.BookField;
import jakarta.validation.Valid;
import jakarta.validation.constraints.NotBlank;

import java.util.List;

public record AuthorResponse(
        @NotBlank String id,
        @NotBlank String lastName,
        @NotBlank String firstName,
        @Valid List<BookField> publishedBooks
) {
}

File: kotlin/src/main/kotlin/fr/sdecout/handson/rest/author/AuthorResponse.kt
package fr.sdecout.handson.rest.author

import fr.sdecout.handson.rest.shared.BookField
import jakarta.validation.Valid
import jakarta.validation.constraints.NotBlank

data class AuthorResponse(
    @field:NotBlank val id: String?,
    @field:NotBlank val lastName: String?,
    @field:NotBlank val firstName: String?,
    @field:Valid val publishedBooks: List<BookField>?,
)
